torch/csrc/autograd/profiler.h

```diff
@@ -202,13 +207,15 @@ struct RangeEventList {
 
   template<typename... Args>
   void record(Args&&... args) {
+    std::lock_guard<std::mutex> guard(mutex_);
     if (blocks.empty() || blocks.front().size() == num_block_elements) {
       allocBlock();
     }
     blocks.front().emplace_back(std::forward<Args>(args)...);
   }
 
   std::vector<Event> consolidate() {
+    std::lock_guard<std::mutex> guard(mutex_);
     std::vector<Event> result;
     for (auto & block : blocks) {
```

Comment:
question, given we added a lock here, will it be faster if we 1) swap/move the blocks, 2) unlock, 3) do the insertion? Looks like `std::forward_list` does have a O(1) move constructor.

> 7) Move constructor. Constructs the container with the contents of other using move semantics. Allocator is obtained by move-construction from the allocator belonging to other.

Reply:
It looks like we call blocks.clear() at the end as well. I think if we move from blocks, then blocks.clear() will be undefined? Although, I think `swap` should work since we can swap it with an empty forward_list.